src/Analyzers/CSharp/Tests/UseParameterNullChecking/UseParameterNullCheckingTests.cs

```diff
@@ -0,0 +1,981 @@
+            {
+                TestCode = testCode,
+                FixedCode = testCode,
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestLocalFunctionCapturingParameter()
+        {
+            var testCode = @"using System;
+class C
+{
+    void M(string param)
+    {
+        local();
+
+        void local()
+        {
+            if (param == null)
+                throw new ArgumentNullException(nameof(param));
+        }
+    }
+}
+";
+            await new VerifyCS.Test()
+            {
+                TestCode = testCode,
+                FixedCode = testCode,
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestNotWithUserDefinedOperator()
+        {
+            var testCode = @"using System;
+class C
+{
+    public static bool operator ==(C c1, C c2) => true;
+    public static bool operator !=(C c1, C c2) => false;
+
+    static void M(C c)
+    {
+        if (c == null)
+            throw new ArgumentNullException(nameof(c));
+    }
+}
+";
+            await new VerifyCS.Test()
+            {
+                TestCode = testCode,
+                FixedCode = testCode,
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestWithUnusedUserDefinedOperator()
+        {
+            await new VerifyCS.Test()
+            {
+                TestCode = @"using System;
+class C
+{
+    public static bool operator ==(C c1, C c2) => true;
+    public static bool operator !=(C c1, C c2) => false;
+
+    static void M(C c)
+    {
+        [|if ((object)c == null)
+            throw new ArgumentNullException(nameof(c));|]
+    }
+}
+",
+                FixedCode = @"using System;
+class C
+{
+    public static bool operator ==(C c1, C c2) => true;
+    public static bool operator !=(C c1, C c2) => false;
+
+    static void M(C c!!)
+    {
+    }
+}
+",
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestWithPreprocessorDirective()
+        {
+            await new VerifyCS.Test()
+            {
+                TestCode = @"using System;
+class C
+{
+    static void M(C c)
+    {
+#nullable enable
+        [|if ((object)c == null)
+            throw new ArgumentNullException(nameof(c));|]
+#nullable disable
+    }
+}
+",
+                FixedCode = @"using System;
+class C
+{
+    static void M(C c!!)
+    {
+#nullable disable
```

Comment:
so this is bad :)

the point of the pp directive is to get us *not* to remove the leading pp directive.  anotehr good case of this would be `#if blah ... #endif`.  if we remove the former we'll end up with broken code.

Reply:
Does an established pattern exist for preserving structured trivia when we're deleting a node?

Reply:
actually, if you are doing SyntaxEditor.RemoveNode, then that should do the right thing.  can you just add the test that has teh `#if`.  i think it should do the right thing, but the test will verify :)